modules/engine/src/utils/tangent.js

```diff
@@ -0,0 +1,64 @@
+import {Vector2, Vector3} from 'math.gl';
+
+const scratchPos0 = new Vector3();
+const scratchPos1 = new Vector3();
+const scratchPos2 = new Vector3();
+
+const scratchUV0 = new Vector2();
+const scratchUV1 = new Vector2();
+const scratchUV2 = new Vector2();
+
+const scratchT1 = new Vector3();
+const scratchT2 = new Vector3();
+const scratchB1 = new Vector3();
+const scratchB2 = new Vector3();
+
+export function calculateTangents(positions, texCoords, indices) {
```

Comment:
With out looking close, since we are not passing normals, I assume they are calculated from triangles. Would it make sense to pass in normals if we have them (to generate bitangents from crossing tangents and normals)

Since it is easy to calculate bitangents in shader, provide option for only calculating tangents?